src/main/scala/dsptools/numbers/chisel_types/FixedPointTypeClass.scala

```diff
@@ -146,17 +147,34 @@ trait FixedPointReal extends FixedPointRing with FixedPointIsReal with Convertab
     ConvertableFromFixedPoint with BinaryRepresentationFixedPoint with RealBits[FixedPoint] with hasContext {
 
   def trimBinary(a: FixedPoint, n: Option[Int]): FixedPoint = {
+    
     // TODO: Support other modes?
     n match {
       case None => a
       case Some(b) => context.trimType match {
+        
         case NoTrim => a
         case Floor => a.setBinaryPoint(b)
         case RoundHalfUp => {
+          
           val roundBp = b + 1
           val addAmt = math.pow(2, -roundBp).F(roundBp.BP)
+          
           plus(a, addAmt).setBinaryPoint(b)
         }
+
+        // This rounds result to the lower or higher value with p0 and p1 probabilities
+        // Unbiased rounding 
+        case StochasticRound => {
+
+          val (cntVal, cntWrap) = Counter (true.B, 16)
+
+          // Add random LSB b bits to the Int number and trim b-1 tail bits
+          val res = (a.asUInt + cntVal(b-1,0)) >> (b-2)
```

Comment:
`chisel3.util` has an LFSR that you may want to use. I'm not sure what your requirements are for correlation of your rounded things, but I think this implementation will result in correlated rounding.